radix-transaction-scenarios/src/scenarios/account_locker.rs

```diff
@@ -0,0 +1,777 @@
+use crate::internal_prelude::*;
+use radix_engine_interface::{
+    blueprints::account::{
+        AccountSetResourcePreferenceInput, ResourcePreference,
+        ACCOUNT_SET_RESOURCE_PREFERENCE_IDENT,
+    },
+    *,
+};
+
+pub struct AccountLockerScenarioConfig {
+    pub account_locker_admin_account: VirtualAccount,
+    pub user_account1: VirtualAccount,
+    pub user_account2: VirtualAccount,
+    pub user_account3: VirtualAccount,
+}
+
+#[derive(Default)]
+pub struct AccountLockerScenarioState {
+    pub(crate) account_locker: State<ComponentAddress>,
+    pub(crate) account_locker_admin_badge: State<ResourceAddress>,
+
+    pub(crate) fungible_resource: State<ResourceAddress>,
+    pub(crate) non_fungible_resource: State<ResourceAddress>,
+}
+
+impl Default for AccountLockerScenarioConfig {
+    fn default() -> Self {
+        Self {
+            account_locker_admin_account: ed25519_account_for_private_key(1),
```

Comment:
I'm afraid we can't just use virtual accounts for this, if we want to be able to run this on e.g. stokenet as part of bottlenose instantiation.

If this isn't being run at genesis, then it's possible for other transactions in the interim to have already used these pre-allocated accounts, which would interfere with the execution of this scenario, causing it to fail.

Setting of things like deposit rules might also interfere with other scenarios.

Instead, we should:
* Create a key pair for each account in the Config
* In the first transaction, use `CREATE_ACCOUNT_ADVANCED`, setting the keys up for the accounts, and capturing the account addresses into state.

Perhaps we should rename `ed25519_account_for_private_key` to `genesis_only_ed25519_account_for_private_key` or something to highlight this for other scenarios!

Reply:
Oh, good point!

I will change the scenario with the change you recommended here. 

The rename sounds good to me and I would also argue that perhaps we should make this an `unsafe` API to force users of this API to read the comments, docs, or at least think about is usage.

Reply:
Thinking about this more, I would like us to add the `deprecated` attribute to the `VirtualAccount` type as a whole and not just the constructor functions with a comment on why it is deprecated and should not be used in future scenarios.

Reply:
Deprecating it also allows us to specify a clear error message to our future selves as seen below:

<img width="935" alt="image" src="https://github.com/radixdlt/radixdlt-scrypto/assets/97489629/bfb5bd28-55f1-41b4-8f50-9957bc8283e6">


Only disadvantage though is that using a deprecated type can get quite annoying with all of the warnings in the old scenarios (which we probably make very little changes to so maybe its fine?).

Reply:
Works for me. We can allow deprecations in the old scenarios 👍